src/components/v2/values/chartValuesDiff/ChartValuesView.tsx

```diff
@@ -0,0 +1,1128 @@
+            }
+        } catch (err) {
+            showError(err)
+            dispatch({
+                type: 'isUpdateInProgress',
+                payload: false,
+            })
+        }
+    }
+
+    const OnEditorValueChange = (codeEditorData: string) => {
+        if (commonState.activeTab !== 'manifest') {
+            dispatch({ type: 'modifiedValuesYaml', payload: codeEditorData })
+        }
+    }
+
+    const redirectToChartValues = async () => {
+        const _chartId = commonState.repoChartValue?.chartId || commonState.installedConfig?.appStoreId
+        if (_chartId) {
+            history.push(getChartValuesURL(_chartId))
+        }
+    }
+
+    const handleTabSwitch = (e) => {
+        if (e?.target && e.target.value !== commonState.activeTab) {
+            if (e.target.value === 'manifest') {
+                const validatedAppName = validationRules.appName(appName)
+                if (!isValidData(validatedAppName)) {
+                    dispatch({
+                        type: 'multipleOptions',
+                        payload: {
+                            openReadMe: false,
+                            openComparison: false,
+                            invalidAppName: !validatedAppName.isValid,
+                            invalidAppNameMessage: validatedAppName.message,
+                            invalidaEnvironment: !commonState.selectedEnvironment,
+                            invalidProject: !commonState.selectedProject,
+                        },
+                    })
+                    toast.error('Please provide the required inputs to view generated manifest')
+                    return
+                }
+            }
+
+            let chartValidationsPayload = {}
+
+            if (
+                commonState.invalidAppName ||
+                commonState.invalidAppNameMessage ||
+                commonState.invalidaEnvironment ||
+                commonState.invalidProject
+            ) {
+                chartValidationsPayload = {
+                    invalidAppName: false,
+                    invalidAppNameMessage: '',
+                    invalidaEnvironment: false,
+                    invalidProject: false,
+                }
+            }
+
+            dispatch({
+                type: 'multipleOptions',
+                payload: {
+                    activeTab: e.target.value,
+                    openReadMe: false,
+                    openComparison: false,
+                    ...chartValidationsPayload,
+                },
+            })
+        }
+    }
+
+    const renderReadMeOption = (disabled?: boolean) => {
+        return (
+            <span
+                className={`chart-values-view__option flex cursor fs-13 fw-6 cn-7 ${
+                    commonState.openReadMe ? 'opened' : ''
+                } ${disabled ? 'disabled' : ''}`}
+                onClick={() => {
+                    if (commonState.fetchingReadMe || disabled) {
```

Comment:
Should we create a onClick function and move this code there?

Reply:
Moved for both readme and comparison.